cmake/vif.cmake

```diff
@@ -4,14 +4,43 @@
 # Generates USB-C VIF policies in XML format from device tree.
 set(gen_vif_script ${ZEPHYR_BASE}/scripts/generate_usb_vif/generate_vif.py)
 set(dts_compatible usb-c-connector)
-set(vif_xml ${PROJECT_BINARY_DIR}/vif.xml)
+set(vif_xml vif.xml)
+set(vif_out ${PROJECT_BINARY_DIR}/${vif_xml})
+
 set(cmd_gen_vif ${PYTHON_EXECUTABLE} ${gen_vif_script}
         --edt-pickle ${EDT_PICKLE}
         --compatible ${dts_compatible}
-        --vif-out ${vif_xml}
-        --board ${BOARD}
+        --vif-out ${vif_out}
         )
 
+if(CONFIG_GENVIF_INPUT_VIF_XML_PATH)
+    if(IS_ABSOLUTE ${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
+        if (EXISTS ${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
+            set(vif_source_xml ${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
+        endif()
+    elseif (EXISTS ${APPLICATION_CONFIG_DIR}/${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
+        set(vif_source_xml ${APPLICATION_CONFIG_DIR}/${CONFIG_GENVIF_INPUT_VIF_XML_PATH})
+    endif()
+else()
+    if (EXISTS ${APPLICATION_SOURCE_DIR}/boards/${BOARD}_${vif_xml})
+        set(vif_source_xml ${APPLICATION_SOURCE_DIR}/boards/${BOARD}_${vif_xml})
+    elseif (EXISTS ${APPLICATION_SOURCE_DIR}/${vif_xml})
+        set(vif_source_xml ${APPLICATION_SOURCE_DIR}/${vif_xml})
```

Comment:
those should also be updated to use `APPLICATION_CONFIG_DIR` instead of `APPLICATION_CONFIG_DIR` to be consistent with other configuration files.

Reply:
Included the changes in new revision